apps/server/src/modules/learnroom/service/course-export.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@nestjs/common';
+import { EntityId } from '@shared/domain';
+import { LessonService } from '@src/modules/lesson/service';
+import { CourseService } from './course.service';
+import { ImsccFileBuilder } from '../imscc';
+
+@Injectable()
+export class CourseExportService {
+	constructor(private readonly courseService: CourseService, private readonly lessonService: LessonService) {}
+
+	async exportCourse(courseId: EntityId): Promise<Buffer> {
+		const course = await this.courseService.findById(courseId);
+		const [lessons] = await this.lessonService.findByCourseIds([courseId]);
+		return new ImsccFileBuilder({
+			title: course.name,
+		})
+			.addOrganizationItems(
+				lessons.map((lesson) => {
+					return {
+						identifier: lesson.id,
```

Comment:
I tried some things, but was not really satisfied with my solutions. So I ended up with a little different strategy. Currently we still use our internal ids, but this can easily be changed. Identifiers are now an argument for every "add" call and can be changed as we see fit in the future.

Reply:
As discussed, the use of the internal id during build up is OK. We may remove those in the generating phase (export). Change OK.